api/src/wfl/executor.clj

```diff
@@ -307,16 +314,17 @@
   "Create new submission from new `source` snapshot if available,
   writing its workflows to `details` table.
   Update statuses for active or failed workflows in `details` table.
-  Return `executor`."
-  [source executor]
+  Return updated `workload`."
+  [{:keys [executor source] :as workload}]
   (when-let [object (stage/peek-queue source)]
-    (let [entity     (from-source executor object)
-          submission (create-submission! executor entity)]
+    (let [entity      (from-source executor object)
+          userComment (create-user-comment "New submission" workload)
+          submission  (create-submission! executor userComment entity)]
       (allocate-submission executor entity submission)
       (stage/pop-queue! source)))
   (update-unassigned-workflow-uuids! executor)
   (update-terra-workflow-statuses! executor)
-  executor)
+  workload)
```

Comment:
This actually doesn't return the updated workload (before it didn't return the updated executor either).  But now we can easily swap with a call to `workloads/load-workload-for-uuid`:

https://github.com/broadinstitute/wfl/blob/develop/api/src/wfl/api/workloads.clj#L70-L79